src/Fleck/WebSocketConnection.cs

```diff
@@ -56,38 +57,38 @@ public WebSocketConnection(ISocket socket, Action<IWebSocketConnection> initiali
       get { return !_closing && !_closed && Socket.Connected; }
     }
 
-    public void Send(string message)
+    public Task Send(string message)
     {
-      Send(message, Handler.FrameText);
+      return Send(message, Handler.FrameText);
     }
 
-    public void Send(byte[] message)
+    public Task Send(byte[] message)
     {
-      Send(message, Handler.FrameBinary);
+        return Send(message, Handler.FrameBinary);
     }
 
-    public void SendPing(byte[] message)
+    public Task SendPing(byte[] message)
     {
-      Send(message, Handler.FramePing);
+        return Send(message, Handler.FramePing);
     }
 
-    public void SendPong(byte[] message)
+    public Task SendPong(byte[] message)
     {
-      Send(message, Handler.FramePong);
+        return Send(message, Handler.FramePong);
     }
 
-    private void Send<T>(T message, Func<T, byte[]> createFrame)
+    private Task Send<T>(T message, Func<T, byte[]> createFrame)
     {
       if (Handler == null)
         throw new InvalidOperationException("Cannot send before handshake");
 
       if (!IsAvailable) {
         FleckLog.Warn("Data sent while closing or after close. Ignoring.");
-        return;
+        return null;

```

Comment:
This should return a faulted task so consumers of Send don't need to do an "if null" check and can choose to handle it through `ContinuationOptions`.